src/amp.js

```diff
@@ -113,30 +100,39 @@ function bootstrap(ampdoc, perf) {
   });
 }
 
-if (shouldMainBootstrapRun) {
-  // Store the originalHash as early as possible. Trying to debug:
-  // https://github.com/ampproject/amphtml/issues/6070
-  if (self.location) {
-    self.location.originalHash = self.location.hash;
-  }
+// Store the originalHash as early as possible. Trying to debug:
+// https://github.com/ampproject/amphtml/issues/6070
+if (self.location) {
+  self.location.originalHash = self.location.hash;
+}
 
-  /** @type {!./service/ampdoc-impl.AmpDocService} */
-  let ampdocService;
-  // We must under all circumstances call makeBodyVisible.
-  // It is much better to have AMP tags not rendered than having
-  // a completely blank page.
-  try {
-    // Should happen first.
-    installErrorReporting(self); // Also calls makeBodyVisibleRecovery on errors.
+/** @type {!./service/ampdoc-impl.AmpDocService} */
+let ampdocService;
+// We must under all circumstances call makeBodyVisible.
+// It is much better to have AMP tags not rendered than having
+// a completely blank page.
+try {
+  // Should happen first.
+  installErrorReporting(self); // Also calls makeBodyVisibleRecovery on errors.
 
-    // Declare that this runtime will support a single root doc. Should happen
-    // as early as possible.
-    installDocService(self, /* isSingleDoc */ true);
-    ampdocService = Services.ampdocServiceFor(self);
-  } catch (e) {
-    // In case of an error call this.
-    makeBodyVisibleRecovery(self.document);
-    throw e;
+  // Declare that this runtime will support a single root doc. Should happen
+  // as early as possible.
+  installDocService(self, /* isSingleDoc */ true);
+  ampdocService = Services.ampdocServiceFor(self);
+} catch (e) {
+  // In case of an error call this.
+  makeBodyVisibleRecovery(self.document);
+  throw e;
+}
+startupChunk(self.document, function initial() {
```

Comment:
Looks like this was duplicated?

Reply:
looks like bad rebase on my part. re-pushed and re-checked with w=1, seems good now. but would appreciate another look from you. thanks